manifests/init.pp

```diff
@@ -4,606 +4,615 @@
-#                                     Defaults based on $::puppetversion
-#                                     type:array
+# $server_ca_auth_required::                Whether client certificates are needed to access the puppet-admin api
+#                                           Defaults to true
+#                                           type:boolean
 #
-# $server_ca_client_whitelist::       The whitelist of client certificates that
-#                                     can query the certificate-status endpoint
-#                                     Defaults to [ '127.0.0.1', '::1', $::ipaddress ]
-#                                     type:array
+# $server_use_legacy_auth_conf::            Should the puppetserver use the legacy puppet auth.conf?
+#                                           Defaults to false (the puppetserver will use its own conf.d/auth.conf)
+#                                           type:boolean
+#
+# $server_check_for_updates::               Should the puppetserver phone home to check for available updates?
+#                                           Defaults to true
+#                                           type:boolean
 #
-# $server_admin_api_whitelist::       The whitelist of clients that
-#                                     can query the puppet-admin-api endpoint
-#                                     Defaults to [ '127.0.0.1', '::1', $::ipaddress ]
-#                                     type:array
-#
-# $server_enable_ruby_profiler::      Should the puppetserver ruby profiler be enabled?
-#                                     Defaults to false
-#                                     type:boolean
-#
-# $server_ca_auth_required::          Whether client certificates are needed to access the puppet-admin api
-#                                     Defaults to true
-#                                     type:boolean
-#
-# $server_use_legacy_auth_conf::      Should the puppetserver use the legacy puppet auth.conf?
-#                                     Defaults to false (the puppetserver will use its own conf.d/auth.conf)
-#                                     type:boolean
-#
-# $server_check_for_updates::         Should the puppetserver phone home to check for available updates?
-#                                     Defaults to true
-#                                     type:boolean
+# $server_environment_class_cache_enabled:: Enable environment class cache
+#                                           Note: If you’re not using Code Manager and opt to enable this setting, make
+#                                           sure your code deployment method – for example r10k – clears the environmen
+#                                           cache when it completes. If you don’t clear the environment cache, the Node
+#                                           Classifier doesn’t receive new class information until Puppet server is
+#                                           restarted.
```

Comment:
I'm unsure if that note text should be really placed here as is, espcially when thinkg about the resulting long output in the foreman-instalelr help and interactive mode. There are many consequences of more or less any switch we offer, still we don't warn at each of these.